libtrip-notification/src/main/java/com/mapbox/navigation/trip/notification/MapboxTripNotificationInterceptor.kt

```diff
@@ -0,0 +1,28 @@
+package com.mapbox.navigation.trip.notification
+
+import androidx.core.app.NotificationCompat
+import com.mapbox.navigation.base.ExperimentalMapboxNavigationAPI
+import com.mapbox.navigation.base.trip.model.RouteProgress
+
+/**
+ * Implement this interface and set it through the [MapboxTripNotification].
+ *
+ * This allows you to extend the notification, or change it to suit your needs. You will need to
+ * use this in order to extend the notification for Android Auto.
+ */
+@ExperimentalMapboxNavigationAPI
+interface MapboxTripNotificationInterceptor {
+    /**
+     * Called when the first notification is being built.
+     */
+    fun interceptFirst(builder: NotificationCompat.Builder): NotificationCompat.Builder {
+        return builder
+    }
+
+    /**
+     * Called periodically to allow you to update the notification with data like [RouteProgress].
+     */
+    fun interceptUpdate(builder: NotificationCompat.Builder): NotificationCompat.Builder {
+        return builder
+    }
```

Comment:
Why do we need to segregate these two calls? Would a client mutate the update differently when it's first or subsequent call?

```suggestion
    fun interceptNotification(builder: NotificationCompat.Builder): NotificationCompat.Builder {
        return builder
    }
```

Reply:
The `MapboxTripNotification` has them segregated, so my thought was if the interceptor wants to create a `MapboxTripNotificationView` it could create it in the `interceptFirst`

But.. not thinking about it, it would be better to use `MapboxNavigationObserver` for that case because then you could also destroy a custom view. 

Updating to a single call